base_import_odoo/models/import_odoo_database.py

```diff
@@ -0,0 +1,472 @@
+            'cronjob_id': self._create_cronjob().id,
+        })
+
+    @api.multi
+    def _run_import(self, commit=True, commit_threshold=100):
+        """Run the import as cronjob, commit often"""
+        self.ensure_one()
+        if not self.password:
+            return
+        # model name: [ids]
+        remote_ids = {}
+        # model name: count
+        remote_counts = {}
+        # model name: count
+        done = {}
+        # mapping_key: local_id
+        idmap = {}
+        # mapping_key: local_id
+        # this are records created or linked when we need to fill a required
+        # field, but the local record is not yet created
+        dummies = {}
+        # model name: [local_id]
+        # this happens when we create a dummy we can throw away again
+        to_delete = {}
+        # dummy_instance
+        dummy_instances = []
+        remote = self._get_connection()
+        self.write({'password': False})
+        if commit and not tools.config['test_enable']:
+            # pylint: disable=invalid-commit
+            self.env.cr.commit()
+        for model_line in self.import_line_ids:
+            model = model_line.model_id
+            remote_ids[model.model] = remote.search(
+                model.model,
+                tools.safe_eval(model_line.domain) if model_line.domain else []
+            )
+            remote_counts[model.model] = len(remote_ids[model.model])
+        self.write({
+            'status_data': {
+                'counts': remote_counts,
+                'ids': remote_ids,
+                'error': None,
+                'done': {},
+            }
+        })
+        if commit and not tools.config['test_enable']:
+            # pylint: disable=invalid-commit
+            self.env.cr.commit()
+        for model_line in self.import_line_ids:
+            model = self.env[model_line.model_id.model]
+            done[model._name] = 0
+
+            for start_index in range(
+                    len(remote_ids[model._name]) / commit_threshold + 1
+            ):
+                index = start_index * commit_threshold
+                ids = remote_ids[model._name][index:index + commit_threshold]
+                context = ImportContext(
+                    remote, model_line, ids, idmap, dummies, dummy_instances,
+                    to_delete, field_context(None, None, None),
+                )
+                try:
+                    self._run_import_model(context)
+                except:
+                    error = traceback.format_exc()
+                    self.env.cr.rollback()
+                    self.write({
+                        'status_data': dict(self.status_data, error=error),
+                    })
+                    # pylint: disable=invalid-commit
+                    self.env.cr.commit()
+                    raise
+                done[model._name] += len(ids)
+                self.write({'status_data': dict(self.status_data, done=done)})
```

Comment:
Why not skip the inner `for start_index` loop, and instead add here an `if` with a checkpoint condition to enter the status data & commit section only when multiples of `commit_threshold` are reached? The effect is the same but (IMO) the code would be simpler and more readable.

Reply:
good point, will add that

Reply:
now as I'm following up on the comments, no, not a good idea. The inner loop is also meant to do the import in blocks (as in: Read a block of data from remote), so we can't really get rid of this loop